codex-rs/core/src/codex.rs

```diff
@@ -346,7 +348,7 @@ pub(crate) struct SessionConfiguration {
     /// Set of feature flags for this session
     features: Features,
     /// Execpolicy policy, applied only when enabled by feature flag.
-    exec_policy: Arc<ExecPolicy>,
+    exec_policy: Arc<RwLock<ExecPolicy>>,
```

Comment:
Hmm, this feels slightly undesirable since `SessionConfiguration` is already generally wrapped in a mutex, but maybe this is OK, I'm not sure yet...